app/src/main/java/com/arygm/quickfix/ui/dashboard/BillsWidget.kt

```diff
@@ -0,0 +1,159 @@
+package com.arygm.quickfix.ui.dashboard
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.platform.testTag // Import for testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.arygm.quickfix.ui.theme.poppinsFontFamily
+import com.arygm.quickfix.ui.theme.poppinsTypography
+
+// Data class for QuickFix item
+data class BillSneakPeak(
+    val name: String,
+    val taskDescription: String,
+    val date: String,
+    val price: Int
+)
+
+@Composable
+fun BillsWidget(
+    billList: List<BillSneakPeak>,
+    onShowAllClick: () -> Unit,
+    onItemClick: (BillSneakPeak) -> Unit,
+    modifier: Modifier = Modifier,
+    itemsToShowDefault: Int = 3
+) {
+  var showAll by remember { mutableStateOf(false) } // Toggle for showing all items
+  BoxWithConstraints {
+    val cardWidth = maxWidth * 0.4f // 40% of the available width for each card
+    val horizontalSpacing = maxWidth * 0.025f // 2.5% of the available width for spacing
+
+    // Card-like styling with shadow for the entire Column
+    Column(
+        modifier =
+            modifier
+                .fillMaxWidth()
+                .padding(horizontalSpacing)
+                .shadow(5.dp, RoundedCornerShape(20.dp))
+                .background(MaterialTheme.colorScheme.surface, RoundedCornerShape(12.dp))
+                .testTag("BillsWidget"), // Added testTag
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.Start) {
+          // Header with divider
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(horizontal = 8.dp),
```

Comment:
This is not an absolute value because the padding is combined with fillMaxWidth(), which is responsive to the screen's width. While the `8.dp` padding itself is a fixed value, its application within a responsive layout makes it part of an adaptive design.

What we should aim to avoid are fixed, absolute-only modifiers like `Modifier.padding(...dp)` on their own or rigid sizes like `Modifier.size(...dp)`, `Modifier.width(...dp)`, and `Modifier.height(...dp)`, as they don't adapt to different screen sizes or densities.

Reply:
I agree with this, sorry for the oversight on my end 😅 you can resolve this